src/test/java/redis/clients/jedis/HostAndPorts.java

```diff
@@ -1,27 +1,28 @@
 package redis.clients.jedis;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 
 public final class HostAndPorts {
 
-  private static List<HostAndPort> redisHostAndPortList = new ArrayList<>();
+  private static HashMap<String, EndpointConfig> endpointConfigs;
+
   private static List<HostAndPort> sentinelHostAndPortList = new ArrayList<>();
   private static List<HostAndPort> clusterHostAndPortList = new ArrayList<>();
   private static List<HostAndPort> stableClusterHostAndPortList = new ArrayList<>();
 
   static {
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 1));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 2));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 3));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 4));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 5));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 6));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 7));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 8));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 9));
-    redisHostAndPortList.add(new HostAndPort("localhost", Protocol.DEFAULT_PORT + 10));
+    String endpointsPath = System.getenv().getOrDefault("REDIS_ENDPOINTS_CONFIG_PATH", "src/test/resources/endpoints.json");
+    try {
+      endpointConfigs = EndpointConfig.loadFromJSON(endpointsPath);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    // TODO: This is a temporary solution to avoid breaking existing tests.
```

Comment:
I would not leave such TODOs in the code. To me this is already huge progress forward, the system remains fully stable and functional, when we get to it we'll do next steps.

Reply:
Makes sense. Removed.